aws/data_source_aws_network_interface.go

```diff
@@ -0,0 +1,174 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsNetworkInterface() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsNetworkInterfaceRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"association": &schema.Schema{
+				Type:     schema.TypeSet,
```

Comment:
Since there's only ever going to be a single association I think this field should be TypeList so it's possible to reference it like `data.aws_network_interface.name.association.0.public_ip` etc.